bindgen/repositories.bzl

```diff
@@ -0,0 +1,49 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("//bindgen/raze:crates.bzl", "raze_fetch_remote_crates")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def maybe(workspace_rule, **kwargs):
+    if not native.existing_rule(kwargs["name"]):
+        workspace_rule(**kwargs)
+
+def rust_bindgen_repositories():
+    """Declare dependencies needed for bindgen."""
+
+    # nb. The bindgen rule itself should work on any platform.
+    _linux_rust_bindgen_repositories()
+
+    raze_fetch_remote_crates()
+
+    native.register_toolchains("@io_bazel_rules_rust//bindgen:example-bindgen-toolchain")
+
+def _linux_rust_bindgen_repositories():
+    # Releases @ http://releases.llvm.org/download.html
+    maybe(
+        http_archive,
+        name = "clang",
```

Comment:
Nit: Is conditionally loading "@clang" preferable to loading our own "@bindgen-clang"?

I feel a bit weird about us trying to fetch "@clang", even if we're doing it nicely by only loading it if it hasn't yet been loaded. I'm curious what the argument is in support of occupying the name "@clang".

Reply:
I have no opinion on this, as I haven't seen other usages of clang (via tarball or otherwise). 
I can scope the name to us, but we should be compatible with any clang executable (would anyone use `@clang` for something else?

Given that the deps management should be through the toolchain instead of providing correctly named external repository, I think shadowing it to `@bindgen_clang` makes sense.